src/prefect/tasks/mysql/mysql.py

```diff
@@ -173,13 +178,24 @@ def run(
                 "The 'fetch' parameter must be one of the following - ('one', 'many', 'all')"
             )
 
+        if cursor_type not in {"cursor", "dict_cursor"}:
+            raise ValueError(
+                f"Unsupported cursor_type '{cursor_type}'. Please choose one of the following - ('cursor', 'dict_cursor')"
+            )
+
+        if cursor_type == "cursor":
+            cursor_class = pymysql.cursors.Cursor
+        else:
+            cursor_class = pymysql.cursors.DictCursor
```

Comment:
Hi! Thanks for contributing to Prefect :) 

I'm curious why you choose to take strings and then do this match rather than just taking the class you're interested in directly? e.g. `cursor_type: pymysql.cursors.Cursor = pymysql.cursors.Cursor`

Reply:
Hey there! Thanks for the prompt response. Good question. My initial thinking was to avoid the user having to import `pymysql.cursors` in their flow. I thought it might be cleaner to reserve all pymysql imports for the pymysql task module, but one could argue that less code wins here, which your suggestion certainly enables. Let me know if you'd like me to make that change.

Reply:
I think that it'd be more maintainable and if you're trying to customize your cursors it seems reasonable to import the option you want. I'd wait until a code owner weighs in though. @jcrist

Reply:
I'm not familiar with `pymysql`. Looks like there's currently 4 types of cursors: https://pymysql.readthedocs.io/en/latest/modules/cursors.html. Perhaps support both? I'd write this as:

```python
cursor_types = {
    "cursor": pymysql.cursors.Cursor,
    "sscursor": pymysql.cursors.SSCursor,
    ...
}
if callable(cursor_type):
    cursor_class = cursor_type
else:
    try:
        cursor_class = cursor_types[cursor_type.lower()]
    except KeyError:
        raise TypeError(f"`cursor_type` should be a callable or one of {'cursor', 'sscursor', 'dictcursor', 'ssdictcursor'}, got {cursor_type}") from None
```

Users can then pass in either a short string name, or a callable class itself. Slight preference for no `_` in the string names, just lowercase the class name.

Reply:
Unless this is a common use case supporting multiple input types for this parameter seems like more work than it's worth, but I also don't have experience with `pymysql` 🤷

Reply:
I deliberately avoided the `SS*` cursors, since these are built to provider iterators for looping through sql results pulled from the SQL server 'on-demand'. E.g.
```python
import pymysql.cursors

conn = pymysql.connect(
host=self.host,
user=self.user,
password=self.password,
db=self.db_name,
charset=self.charset,
port=self.port,
cursorclass=pymysql.cursors.SSCursor
)

with conn.cursor() as cursor:
    cursor.execute(query)

    for row in cursor:
        # each iteration fetches a new result from the sql server, as opposed to all results buffered in memory at once.
        print(row)
```

I'm skeptical that this paradigm has a place in this task class (or if it's generally a broader Prefect anti-pattern), since as of now we're just fetching one/some/all results and returning at once.

Reply:
Makes sense to me. I'm fine with the shorthand names, provided:
- the names match the class names sans capitalization (no underscore, just check `name.lower()` matches `cursor` or `dictcursor`)
- You can also pass in a class directly
- We check the full set and error on unknown names (right now if it's not `cursor` it's `dictcursor`, even if the user types something else).

Reply:
Thanks @jcrist. I'll have another commit pushed by tomorrow.

Reply:
@jcrist @madkinsz I implemented the changes discussed above. Let me know what you think.

Reply:
@mhmcdonald thanks again for taking the time to work on this! I've marked a few conversations about type annotations as resolved as I don't think it's worth fretting over unless you wanted to. Following a couple edge case fixes this looks great.